lib/web_ui/test/text_editing_test.dart

```diff
@@ -549,6 +549,41 @@ void main() {
       expect(spy.messages, isEmpty);
     });
 
+    test('negative base and offset values in editing state is handled', () {
+      final MethodCall setClient = MethodCall(
+          'TextInput.setClient', <dynamic>[123, flutterSinglelineConfig]);
+      textEditing.handleTextInput(codec.encodeMethodCall(setClient));
+
+      const MethodCall setEditingState1 =
+          MethodCall('TextInput.setEditingState', <String, dynamic>{
+        'text': 'abcd',
+        'selectionBase': 2,
+        'selectionExtent': 3,
+      });
+      textEditing.handleTextInput(codec.encodeMethodCall(setEditingState1));
+
+      const MethodCall show = MethodCall('TextInput.show');
+      textEditing.handleTextInput(codec.encodeMethodCall(show));
```

Comment:
I'd call `checkInputEditingState` before sending `setEditingState2` to make sure that sending it had an effect at all. Otherwise, what if due to some weird bug it is always set to 0?

Reply:
Done.